pallets/acurast-xcm-sender/src/lib.rs

```diff
@@ -9,82 +9,84 @@ mod tests;
-        // - and the remaining bytes are the extrinsic parameter SCALE encoded.
-        encoded_call.append(&mut call.encode());
-
-        // Add transact instruction with the fulfill call.
-        let transaction = Transact {
-            origin_type: OriginKind::Xcm,
-            require_weight_at_most: 1_000_000_000u64, // TODO: Review this value
-            call:encoded_call.into(),
-        };
-
-        // Submit the xcm message
-        match T::XcmSender::send_xcm(
-            (1, X1(Parachain(T::AcurastParachainId::get()))),
-            Xcm(vec![transaction]),
-        ) {
-            Ok(()) => {
-                Pallet::<T>::deposit_event(Event::XcmSent { call, caller });
-                Ok(())
-            }
-            Err(error) => {
-                Pallet::<T>::deposit_event(Event::XcmNotSent {
-                    call,
-                    caller,
-                    error,
-                });
-                Err(Error::<T>::XcmError.into())
-            }
+    fn split_multi_location<T: Config>(
+        multi_location: MultiLocation,
+    ) -> Result<(MultiLocation, u8), Error<T>> {
+        match multi_location.split_last_interior() {
+            (multi_location, Some(Junction::PalletInstance(value))) => Ok((multi_location, value)),
+            _ => Err(Error::<T>::InvalidDestination),
         }
     }
 
     #[pallet::pallet]
-    #[pallet::generate_store(pub(super) trait Store)]
     pub struct Pallet<T>(_);
 
+    impl<T: Config> Pallet<T> {
+        pub fn send(
+            caller: T::AccountId,
+            destination: MultiLocation,
+            payload: Vec<u8>,
+            parameters: Option<Vec<u8>>,
+        ) -> DispatchResult {
+            let (xcm_destination, pallet_instance) = split_multi_location::<T>(destination)?;
+            // Encode extrinsic call version of the call
+            // - first byte is the pallet id on the destination chain;
+            let mut encoded_call = vec![pallet_instance];
+            // - second byte is the position of the extrinsic function being called;
+            // - and the remaining bytes are the extrinsic parameter SCALE encoded.
+            let call = ReceiverCall::Fulfill(payload, parameters);
+            encoded_call.append(&mut call.encode());
+
+            // Add transact instruction with the fulfill call.
+            let transaction = Transact {
+                origin_type: OriginKind::Xcm,
+                require_weight_at_most: 1_000_000_000u64, // TODO: Review this value
+                call: encoded_call.into(),
+            };
+
+            // Submit the xcm message
+            match T::XcmSender::send_xcm(xcm_destination, Xcm(vec![transaction])) {
+                Ok(()) => {
+                    Pallet::<T>::deposit_event(Event::XcmSent { call, caller });
+                    Ok(())
+                }
+                Err(error) => {
+                    Pallet::<T>::deposit_event(Event::XcmNotSent {
```

Comment:
This event will be roll-backed. I will work on improving the event handling on a next task